OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs

```diff
@@ -830,6 +839,23 @@ static void ShowStatusBarsDropdown(DropDownButtonWidget dropdown, GameSettings s
 			dropdown.ShowDropDown("LABEL_DROPDOWN_TEMPLATE", 500, options.Keys, setupItem);
 		}
 
+		static void ShowDisplaySelectionDropdown(DropDownButtonWidget dropdown, GraphicSettings s)
+		{
+			var options = Game.Renderer.Displays;
+
+			Func<string, ScrollItemWidget, ScrollItemWidget> setupItem = (o, itemTemplate) =>
+			{
+				var item = ScrollItemWidget.Setup(itemTemplate,
+					() => s.VideoDisplay == options[o],
+					() => s.VideoDisplay = options[o]);
+
+				item.Get<LabelWidget>("LABEL").GetText = () => o;
```

Comment:
Please use something like
```csharp
var font = Game.Renderer.Fonts[itemTemplate.Font];
var label = WidgetUtils.TruncateText(o, itemTemplate.Bounds.Width, font);
item.Get<LabelWidget>("LABEL").GetText = () => label
```
to protect against long display names.

We'll need to do a similar cut (using `CachedTransform` - there are several examples elsewhere in the UI code for this) for the selected dropdown display.

Reply:
Probably better to use "Display N" here or something. Cant tell which display is which :)
![display](https://user-images.githubusercontent.com/19173272/73935598-2c855b00-48e1-11ea-9dc8-3c0ab376d1fa.png)

Reply:
macOS gives the actual display model (e.g U2718Q), but if this is what we have to deal with under windows then I completely agree about changing these to "Display N".

Reply:
Linux also gives back proper names for the displays. So which version would you like? Naming used here in the comments also raises the question whether it should be called "display" or "monitor".

Reply:
~70% of players are on Windows, so lets go with either Monitor or Display. MacOS and I think Windows (someone correct me if i'm wrong) uses Display, so I think we should go with that for consistency.

Reply:
> Please use something like
> 
> ```cs
> var font = Game.Renderer.Fonts[itemTemplate.Font];
> var label = WidgetUtils.TruncateText(o, itemTemplate.Bounds.Width, font);
> item.Get<LabelWidget>("LABEL").GetText = () => label
> ```
> 
> to protect against long display names.
> 
> We'll need to do a similar cut (using `CachedTransform` - there are several examples elsewhere in the UI code for this) for the selected dropdown display.

I think this is resolved due to the simpler approach with a constant string which cannot result in options that are too long.